src/metabase/api/collection.clj

```diff
@@ -101,7 +101,14 @@
                        (perms/audit-namespace-clause :namespace namespace)
                        (collection/visible-collection-ids->honeysql-filter-clause
                         :id
-                        (collection/permissions-set->visible-collection-ids permissions-set))]
+                        (collection/permissions-set->visible-collection-ids
+                         permissions-set
+                         {:include-archived (if archived
+                                              :only
+                                              :exclude)
```

Comment:
Yeah :/

Initially I wanted to rename everything to do with `archive` to `trash`. However, I didn't want to hassle with additional migrations and the code churn required to actually change the column names (or the API) over. So that inconsistency is just there - although honestly, maybe it would be better to just use the word "Trash" on the frontend, and have the backend consistently refer to "The archive" or "archived" items. 🤔 

Then, in this case, it's worse, because I let the inconsistent column name bleed into more code. So really, I suppose this should be `include-trashed` rather than `include-archived`... even though it's checking the `:archived` column? 😭 

To answer your question about why we need both `include-archived` and `include-trash?` - `include-trash?` is whether we should include the Trash Collection itself, and `include-archived` is whether we should include things that are "in" the trash.

Reply:
I'm personally in favor of consistent naming even if it requires migrations/a little more code churn. Otherwise we built up a bank of implicit historical knowledge you need to figure out to approach the codebase. 

But if it's too much to do as part of this PR, maybe open a tech debt issue for it? At least it'll be documented and could be a good onboarding task in the future.

For the naming of these flags, maybe `include-archived-items?` and `include-trash-collection?`?

Reply:
Renamed to `:include-archived-items` and `:include-trash-collection?`, and replaced `trashed` with `archived` everywhere, including in migrations (e.g. `trashed_directly` => `archived_directly` and `trash_operation_id` => `archive_operation_id`)